### PR TITLE
Naming scheme to simulation geometry / refactor TOF DPL digitizer

### PR DESCRIPTION
**Pull request by sawenzel**
Instead of always writing O2geometry.root we now follow the same scheme as for hits and kinematics: The geometry file will be `prefix_geometry.root`.

Also refactoring the TOFDigitizerSpec to use the DPL-Task formalism.

### CODE REVIEW

File: DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/NameConf.h [hunk 1]
```diff
@@ -54,12 +54,26 @@ class NameConf : public o2::conf::ConfigurableParamHelper<NameConf>
...
  private:
+  // unmodifiable constants used to construct filenames etc
   static constexpr std::string_view STANDARDSIMPREFIX = "o2sim";
   static constexpr std::string_view HITS_STRING = "Hits";     // hardcoded
   static constexpr std::string_view DIGITS_STRING = "Digits"; // hardcoded
   static constexpr std::string_view GRP_STRING = "grp";       // hardcoded
   static constexpr std::string_view KINE_STRING = "Kine";     // hardcoded
+  static constexpr std::string_view GEOM_STRING = "geometry";
 
   O2ParamDef(NameConf, "NameConf");
```
Comment (shahor02) [inline DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/NameConf.h]: Given that simulation anyway must provide external prefixes, perhaps we indeed don't make this class ConfigurableParam at all, making all names constexpr. Instead, I would make all defaults for the prefix arguments = ``""``, the ``o2sim`` can be added internally, so the non-trivial prefix (or path) would come on top of it.

The reason is that if in the reconstruction I want to take the data (and geometry) from the path ``string XXX``, I don't want to call it as ``NameConf::getGeomFileName( XXX + ' /' + NameConf::STANDARDSIMPREFIX)``,
but simply ``NameConf::getGeomFileName(XXX)``

File: macro/CheckClusters_mft.C [hunk 2]
```diff
@@ -53,7 +53,8 @@ void CheckClusters_mft(Int_t nEvents = 10, Int_t nMuons = 200)
   Char_t filename[100];
 
   // Geometry
-  o2::base::GeometryManager::loadGeometry("O2geometry.root", "FAIRGeom");
+  sprintf(filename, NameConf::getGeomFileName("o2sim").c_str());
```
Comment (shahor02) [inline macro/CheckClusters_mft.C]: I think we can add ``FAIRGeom`` to NameConf, and change the ``o2::base::GeometryManager::loadGeometry`` to
````
static void loadGeometry(std::string_view geomFileName = "")
{
   std::string name = geomFileName.empty ? NameConf::getGeomFileName() : geomFileName();
   TFile flGeom( name.c_str() );
   ...
}
````

File: DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/NameConf.h (hunk 1 above)
Comment (sawenzel) [reply DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/NameConf.h]: Ok. I'll modify it this way.

File: macro/CheckClusters_mft.C (hunk 2 above)
Comment (sawenzel) [reply macro/CheckClusters_mft.C]: yes absolutely

**Comment by sawenzel** [comment]
@noferini: Could you please take a look if the changes to TOF are ok? I tested it locally and it gives identical results. This change is good as now all digitizers follow the task scheme and we can go on and develop a little framework for all digitizers to reduce boilerplate code.

**Pull request merged**